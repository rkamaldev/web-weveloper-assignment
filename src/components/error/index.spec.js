/* eslint-disable no-undef */
import React from 'react';
import * as nextRouter from 'next/router';
import renderer from 'react-test-renderer';

import Error from './';

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    CONTACT_EMAIL: 'test@test.com',
  },
}));

describe('#Error', () => {
  it('Should render error component with support email id', () => {
    // eslint-disable-next-line no-import-assign
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: '/' }));
    const component = renderer.create(<Error />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
