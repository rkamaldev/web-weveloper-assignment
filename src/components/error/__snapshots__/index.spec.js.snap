// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Error Should render error component with support email id 1`] = `
<div
  className="body"
>
  <div
    className="left"
  >
    <div
      className="wrapper"
    >
      <div
        className="title"
      >
        Oops, looks like something went wrong!!!
      </div>
      <div
        className="description"
      >
        Please try again later or contact 
        <span>
          test@test.com
        </span>
      </div>
      <div
        className="goHomeButtonWrapper"
      >
        <button
          className="goHomeButton"
          onClick={[Function]}
        >
          Back to Home
        </button>
      </div>
    </div>
  </div>
  <div
    className="right"
  >
    <img
      alt="me"
      src="/error-page.png"
      width="80%"
    />
  </div>
</div>
`;

exports[`Should render error component with support email id 1`] = `
<div
  className="body"
>
  <div
    className="left"
  >
    <div
      className="wrapper"
    >
      <div
        className="title"
      >
        Oops, looks like something went wrong!!!
      </div>
      <div
        className="description"
      >
        Please try again later or contact 
        <span>
          test@test.com
        </span>
      </div>
      <div
        className="goHomeButtonWrapper"
      >
        <button
          className="goHomeButton"
          onClick={[Function]}
        >
          Back to Home
        </button>
      </div>
    </div>
  </div>
  <div
    className="right"
  >
    <img
      alt="me"
      src="/error-page.png"
      width="80%"
    />
  </div>
</div>
`;
